File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s==" ") return true;
        string s1="";
        for(char i:s){
            i=tolower(i);
            if((i>=97&&i<=122)||i>=48&&i<=57) s1+=i;
        }
        if(s1.size()==1) return true;
        int i=0,j=s1.size()-1;
        while(i<=j){
            if(s1[i]!=s1[j]) return false;
            i++;
            j--;
        }
        return true;
    }
};

File: string/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
      String s1="";
      for(int i=0;i<s.length();i++){
        if(Character.isLetterOrDigit(s.charAt(i))) s1+=s.charAt(i);
      }  
      s1=s1.toLowerCase();
      int i=0;
      int j=s1.length()-1;
      System.out.println(s1);
      while(i<=j){
        if(s1.charAt(i)!=s1.charAt(j)) return false;
        i++;
        j--;
      }
      return true;
    }
}